crates/sui-source-validation/src/lib.rs

```diff
@@ -610,12 +611,19 @@ fn download_and_compile(
 ) -> anyhow::Result<()> {
     let dest_dir = PathBuf::from_iter([&*MOVE_HOME, "binaries"]); // E.g., ~/.move/binaries
     let dest_version = dest_dir.join(compiler_version);
-    let mut dest_canonical_binary = dest_version.clone();
-    dest_canonical_binary.extend(["target", "release", CANONICAL_BINARY_NAME]);
+    let mut dest_canonical_path = dest_version.clone();
+    dest_canonical_path.extend(["target", "release"]);
+    let mut dest_canonical_binary = dest_canonical_path.clone();
+
+    let platform = detect_platform(&root, compiler_version, &dest_canonical_path)?;
+    if platform == "windows-x86_64" {
+        dest_canonical_binary.push(CANONICAL_WIN_BINARY_NAME);
+    } else {
+        dest_canonical_binary.push(CANONICAL_UNIX_BINARY_NAME);
+    }
 
     if !dest_canonical_binary.exists() {
         // Check the platform and proceed if we can download a binary. If not, the user should follow error instructions to sideload the binary.
-        let mut platform = detect_platform(&root, compiler_version, &dest_canonical_binary)?;
```

Comment:
With this change we have to `detect_platform` before checking if the binary exists. 

There's a method here close to the previous check where we can do "does binary exist?" by checking whether either `sui` or `sui.exe` exists. But with this PR though, we have to reuse knowledge of the platform to also _invoke_ either `sui` or `sui.exe` depending on the platform. So I think it's more natural to get the `platform` and have the logic case out on that, than guess and persist `sui` vs `sui.exe`